app/models/miq_worker/runner.rb

```diff
@@ -335,41 +338,35 @@ def heartbeat
     do_exit("Error heartbeating because #{err.class.name}: #{err.message}\n#{err.backtrace.join('\n')}", 1)
   end
 
-  def heartbeat_to_drb
-    # Disable heartbeat check.  Useful if a worker is running in isolation
-    # without the oversight of MiqServer::WorkerManagement
-    return if skip_heartbeat?
-
+  def process_messages_from_server
     worker_monitor_drb.register_worker(@worker.pid, @worker.class.name, @worker.queue_name)
-    worker_monitor_drb.update_worker_last_heartbeat(@worker.pid)
-
     worker_monitor_drb.worker_get_messages(@worker.pid).each do |msg, *args|
       process_message(msg, *args)
     end
   rescue DRb::DRbError => err
-    do_exit("Error heartbeating to MiqServer because #{err.class.name}: #{err.message}", 1)
+    do_exit("Error processing messages from MiqServer because #{err.class.name}: #{err.message}", 1)
```

Comment:
if this PR is about removing DRb-based heartbeating, why do we have `rescue DRb::DRbError` here?

Reply:
This method in particular is not for heartbeat. This fetches messages set for the worker in the `@workers` hash over drb.

The vim broker is the only worker that still uses this functionality (previously it was also used for config sync and stopping workers), so this can all be removed once the vim broker is gone.

Reply:
To clarify, processing messages and heartbeating are two different operations that we happen to execute at the same time.

Reply:
thanks for the clarification @carbonin